### DISCUSSION TITLE
Anyone tried to use SWC with Remix?

### DISCUSSION THREAD

**Discussion by kawpii** [remix-run/remix#1323] [Proposals]
SWC : https://github.com/swc-project/swc

**Comment by sergiodxa** [comment]
Remix already had a compiler which is really fast and not replaceable, why would you want to use swc? If you really have a use case you could put your code in some src folder, use swc to compile it and store the output in the app folder, but that will make your build slower since you have two builds now

**Reply by kawpii** [reply]
> Remix already had a compiler

So they transpile, bundle code using their own compiler, I thought they were using Rollup or Webpack?

**Reply by sergiodxa** [reply]
They use esbuild, with some custom configuration, they don’t use webpack or rollup (they used this one while in beta before switching to esbuild)

And esbuild is really really fast

**Comment by parisholley** [comment]
yes, i've been patching remix inside of `@remix-run/dev` to add the following in the `compilerServer.js`/`compilerBrowser.js`:

```typescript
  plugins.unshift(require('esbuild-plugin-decorator').esbuildPluginDecorator({
      tsconfigPath: 'tsconfig.json',compiler: 'swc'
  }));
```
I use frameworks which rely on decorators (ORM Mapping), have had zero issues, running in prod

**Comment by shaqq** [comment]
Hello - my app is currently a Vite + SWC app. I'm trying to move to Remix + Vite.

One motivation of using SWC is the existence of plugins that integrate with React Refresh, like this one from jotai https://jotai.org/docs/tools/swc

Is it possible to do anything similar, functionally, with Remix?
